### PR TITLE
Allow specification of project branch on creation

### PR DESCRIPTION
**Pull request by kruppel**
Adds a text field for specifying project branch to project creation form. Defaults to master.

For processing webhook events and building projects, the project setting will be used to determine which branch to deploy.

image

### CODE REVIEW

File: server/plugins/codeflow/codeflow.go [hunk 1]
```diff
@@ -299,6 +295,19 @@ func (x *Codeflow) Process(e agent.Event) error {
 			return err
 		}
 
+		// Only build project branch
+		branch := project.GitBranch
+
+		if len(strings.TrimSpace(branch)) == 0 {
```
Comment (stewartpark) [inline server/plugins/codeflow/codeflow.go]: Can `branch` be nil?

Comment (kruppel) [inline server/plugins/codeflow/codeflow.go]: Nope. In golang, for a struct, if the value is a string, when unset, it'll be an empty string.

https://play.golang.org/p/yR_NSAo7TH

**Comment by jeremyd** [comment]
Sorry I didn't see this earlier, is this ready to ship?

**Comment by kruppel** [comment]
> Sorry I didn't see this earlier, is this ready to ship?

@jeremyd I talked with @sasso about some changes to make this project-based (i.e. one project, multiple environments). I haven't gotten to the changes just yet, but you can be sure I'll reach out once there's something ready.

**Pull request closed without merging**